designer_v2/lib/features/app_drawer.dart

```diff
@@ -81,55 +90,67 @@ class AppDrawer extends ConsumerStatefulWidget {
-      ),
-    ],
-    [
-      GoRouterDrawerEntry(
-        localizedTitle: () => tr.navlink_public_studies,
-        icon: Icons.public,
-        intent: RoutingIntents.publicRegistry,
-        localizedHelpText: () => tr.navlink_public_studies_tooltip,
-      ),
-    ]
-  ];
+  late final List<List<GoRouterDrawerEntry>> topEntries;
 
   /// List of sections with their corresponding menu entries
-  final List<List<DrawerEntry>> bottomEntries = [
-    [
-      DrawerEntry(
-          localizedTitle: () => tr.navlink_account_settings,
-          icon: Icons.settings_rounded,
-          onSelected: (context, ref) {
-            showDialog(context: context, builder: (context) => const AccountSettingsDialog());
-          }),
-      DrawerEntry(
-        localizedTitle: () => tr.navlink_logout,
-        icon: Icons.logout_rounded,
-        onSelected: (context, ref) {
-          ref.read(authRepositoryProvider).signOut();
-        },
-      ),
-    ],
-  ];
+  late final List<List<DrawerEntry>> bottomEntries;
 
   List<DrawerEntry> get allEntries => [...topEntries, ...bottomEntries].expand((e) => e).toList();
 
   /// Index of the currently selected [[NavigationGoRouterEntry]]
   /// Defaults to -1 if none of the entries is currently selected
   int _selectedIdx = -1;
 
+  @override
+  void initState() {
+    super.initState();
+    topEntries = [
+      [
+        GoRouterDrawerEntry(
+          localizedTitle: () => tr.navlink_my_studies,
+          autoCloseDrawer: widget.autoCloseDrawer,
+          icon: Icons.folder_copy_rounded,
+          intent: RoutingIntents.studies,
+        ),
+        GoRouterDrawerEntry(
+          localizedTitle: () => tr.navlink_shared_studies,
+          autoCloseDrawer: widget.autoCloseDrawer,
+          icon: Icons.folder_shared_rounded,
+          intent: RoutingIntents.studiesShared,
+          enabled: false,
+        ),
+      ],
+      [
+        GoRouterDrawerEntry(
+          localizedTitle: () => tr.navlink_public_studies,
+          autoCloseDrawer: widget.autoCloseDrawer,
+          icon: Icons.public,
+          intent: RoutingIntents.publicRegistry,
+          localizedHelpText: () => tr.navlink_public_studies_tooltip,
+        ),
+      ]
+    ];
+    bottomEntries = [
+      [
+        DrawerEntry(
+            localizedTitle: () => tr.navlink_account_settings,
+            autoCloseDrawer: widget.autoCloseDrawer,
+            icon: Icons.settings_rounded,
+            onSelected: (context, ref) {
+              showDialog(context: context, builder: (context) => const AccountSettingsDialog());
+            }),
+        DrawerEntry(
+          localizedTitle: () => tr.navlink_logout,
+          autoCloseDrawer: widget.autoCloseDrawer,
+          icon: Icons.logout_rounded,
+          onSelected: (context, ref) {
+            ref.read(authRepositoryProvider).signOut();
+          },
+        ),
+      ],
+    ];
+  }
+
   @override
   void didUpdateWidget(AppDrawer oldWidget) {
```

Comment:
Commit b01544bd breaks the update of the selected route.
That's why I reverted it. 

@johannesvedder : Do you have a better idea for the update of the selected route that works with a `const AppDrawer`?